pkg/provider/kubernetes/auth.go

```diff
@@ -109,35 +109,17 @@ func (k *BaseClient) setClientCert(ctx context.Context) error {
 	return nil
 }
 
-func (k *BaseClient) secretKeyRefForServiceAccount(ctx context.Context, serviceAccountRef *esmeta.ServiceAccountSelector) ([]byte, error) {
-	serviceAccount := &corev1.ServiceAccount{}
-	ref := types.NamespacedName{
-		Namespace: k.namespace,
-		Name:      serviceAccountRef.Name,
-	}
+func (k *BaseClient) serviceAccountToken(ctx context.Context, serviceAccountRef *esmeta.ServiceAccountSelector) ([]byte, error) {
```

Comment:
Kubernetes provider will migrate to the new TokenRequest API. I don't see a need to keep the old `ServiceAccount.Secrets[]` approach.